src/main/java/graphql/schema/idl/SchemaGeneratorDirectiveHelper.java

```diff
@@ -0,0 +1,149 @@
+package graphql.schema.idl;
+
+import graphql.language.NamedNode;
+import graphql.language.NodeParentTree;
+import graphql.schema.GraphQLArgument;
+import graphql.schema.GraphQLDirective;
+import graphql.schema.GraphQLDirectiveContainer;
+import graphql.schema.GraphQLEnumType;
+import graphql.schema.GraphQLEnumValueDefinition;
+import graphql.schema.GraphQLFieldDefinition;
+import graphql.schema.GraphQLInputObjectField;
+import graphql.schema.GraphQLInputObjectType;
+import graphql.schema.GraphQLInterfaceType;
+import graphql.schema.GraphQLObjectType;
+import graphql.schema.GraphQLScalarType;
+import graphql.schema.GraphQLUnionType;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static graphql.Assert.assertNotNull;
+
+class SchemaGeneratorDirectiveHelper {
+
+    static class Parameters {
+        private final TypeDefinitionRegistry typeRegistry;
+        private final RuntimeWiring runtimeWiring;
+        private final NodeParentTree nodeParentTree;
+        private final Map<String, Object> context;
+
+        Parameters(TypeDefinitionRegistry typeRegistry, RuntimeWiring runtimeWiring, NodeParentTree<NamedNode> nodeParentTree, Map<String, Object> context) {
+            this.typeRegistry = typeRegistry;
+            this.runtimeWiring = runtimeWiring;
+            this.nodeParentTree = nodeParentTree;
+            this.context = context;
+        }
+
+        public TypeDefinitionRegistry getTypeRegistry() {
+            return typeRegistry;
+        }
+
+        public RuntimeWiring getRuntimeWiring() {
+            return runtimeWiring;
+        }
+
+        public NodeParentTree<NamedNode> getNodeParentTree() {
+            return nodeParentTree;
+        }
+
+        public Map<String, Object> getContext() {
+            return context;
+        }
+    }
+
+    public GraphQLObjectType onObject(GraphQLObjectType element, Parameters params) {
```

Comment:
I think we should unit test each of the possible locations for directive wirings.

Reply:
I do.  see `def "will trace down into each directive callback"() {`

It asserts that each call back has a certain shape and that they are called at all.

It exercises all the known callbacks